### PR TITLE
dictionaryish(NULL) return true

### PR DESCRIPTION
**Pull request by ilovemane**
#1712

### CODE REVIEW

File: R/attr.R [hunk 1]
```diff
@@ -110,11 +110,16 @@ detect_void_name <- function(x) {
 is_dictionaryish <- function(x) {
   # 2022-01: Used in many packages. Don't deprecate without a
   # replacement.
-  if (!length(x)) {
-    return(!is.null(x))
-  }
+  if (is.null(x)) {
+    TRUE
```
Comment (lionel-) [inline R/attr.R:114]: I'd do this with an early return so that the rest of the function is not nested in a branch. So something like:

```r
if (is.null(x)) {
  return(TRUE)
}
if (!length(x)) {
  return(!is.null(x))
}
```

File: R/attr.R [hunk 2]
```diff
@@ -110,11 +110,16 @@ detect_void_name <- function(x) {
...
+  } else {
+    if (!length(x)) {
+      return(!is.null(x))
+    }
 
-  is_named(x) && !any(duplicated(names(x)))
+    is_named(x) && !any(duplicated(names(x)))
+
```
Comment (lionel-) [inline R/attr.R:121]: Can you remove this new unnecessary whitespace please?

**Review by lionel- — COMMENTED** [review]
Looking good! Just a couple of nitpicks.

Also can you please add a NEWS bullet referencing the issue number and tagging your username?

File: R/attr.R (hunk 2 above)
Comment (ilovemane) [reply R/attr.R:116]: done!

File: R/attr.R (hunk 1 above)
Comment (ilovemane) [reply R/attr.R]: done!

**Comment by lionel-** [comment]
Thank you!
